Angular 17 SSR deployment does not work on windows

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

<!--
Thank you for contributing to the Firebase community!

Think you found a bug?
=======================
Yeah, we're definitely not perfect! Please use this template and include a minimal repro when opening the issue. If you know how to solve the issue, please create a Pull Request, and we'd be happy to review it!

Have a feature request?
========================
Great, we love hearing how we can improve our products! However, do not use this template to submit a feature request. Please submit your feature requests to: https://firebase.google.com/support/contact/bugs-features/

Have a usage question?
=======================
We get lots of those and we love helping you, but GitHub is not the best place for them and they will be closed. Please take a look at the guide first: https://firebase.google.com/docs/cli/

If the official documentation doesn't help, try asking through our official support channel: https://firebase.google.com/support/

Additional locations to check for solutions or assistance from the community:
- Stack Overflow: https://stackoverflow.com/
- Firebase Slack Community: https://firebase.community/

*Please avoid duplicate posting across multiple channels!*
-->

### [REQUIRED] Environment info

<!-- What version of the Firebase CLI (`firebase-tools`) are you using? Note that your issue may already be fixed in the latest versions. The latest version can be found at https://github.com/firebase/firebase-tools/releases -->

<!-- Output of `firebase --version` -->

**firebase-tools:** The current `master` branch with Commit https://github.com/firebase/firebase-tools/commit/721c33015c3b160570a4af1a916da77797e2e991

<!-- e.g. macOS, Windows, Ubuntu -->

**Platform:** Windows

### [REQUIRED] Test case

- Generate a new Angular 17 SSR enabled application
- Setup firebase deployment with the web frameworks feature
- Add `"firebase-tools": "firebase/firebase-tools#master",` to your `package.json` and run `npm install` to use the current `master` branch

### [REQUIRED] Steps to reproduce

- Run `npx firebase deploy` on Windows
- Open the browser and access our deployed app

### [REQUIRED] Expected behavior

Running `npx firebase deploy` on Windows generates a valid Angular SSR function.

### [REQUIRED] Actual behavior

Look at the Google Cloud Function Logs and find the following error. Note the slashes are missing. The required module is actually located at `/workspace/dist/your-project-name/server/server.mjs`
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/workspace/distyour-project-nameserver/server.mjs' imported from /workspace/bootstrap.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:524:22)
    at importModuleDynamically (node:internal/modules/cjs/loader:1188:29)
    at importModuleDynamicallyWrapper (node:internal/vm/module:429:21)
```

To verify this look at the `bootstrap.js` file in the `.firebase/<your-project-id>/functions` and see the path to the `server.mjs` 
 contains backslashes: `./dist\your-project-name\server/server.mjs`

--- a/src/frameworks/angular/utils.ts
+++ b/src/frameworks/angular/utils.ts
@@ -5,7 +5,7 @@ import type { WorkspaceNodeModulesArchitectHost } from "@angular-devkit/architec
 import { AngularI18nConfig } from "./interfaces";
 import { relativeRequire, validateLocales } from "../utils";
 import { FirebaseError } from "../../error";
-import { join } from "path";
+import { join, posix, sep } from "path";
 import { BUILD_TARGET_PURPOSE } from "../interfaces";
 import { AssertionError } from "assert";
 import { assertIsString } from "../../utils";
@@ -420,7 +420,9 @@ export async function getServerConfig(sourceDir: string, configuration: string) 
   }
   const browserTargetOptions = await architectHost.getOptionsForTarget(buildOrBrowserTarget);
   assertIsString(browserTargetOptions?.outputPath);
-  const browserOutputPath = join(browserTargetOptions.outputPath, buildTarget ? "browser" : "");
+  const browserOutputPath = join(browserTargetOptions.outputPath, buildTarget ? "browser" : "")
+    .split(sep)
+    .join(posix.sep);
   const packageJson = JSON.parse(await host.readFile(join(sourceDir, "package.json")));
 
   if (!ssr) {
@@ -449,7 +451,9 @@ export async function getServerConfig(sourceDir: string, configuration: string) 
   );
   const serverTargetOptions = await architectHost.getOptionsForTarget(buildOrServerTarget);
   assertIsString(serverTargetOptions?.outputPath);
-  const serverOutputPath = join(serverTargetOptions.outputPath, buildTarget ? "server" : "");
+  const serverOutputPath = join(serverTargetOptions.outputPath, buildTarget ? "server" : "")
+    .split(sep)
+    .join(posix.sep);
   if (serverLocales && !defaultLocale) {
     throw new FirebaseError(
       "It's required that your source locale to be one of the localize options"

